design/www/app/chatpage/chatpage.controller.js

```diff
@@ -4,39 +4,49 @@
      angular.module('starter')
          .controller('chatPageController', chatPageController);
 
-     function chatPageController($scope,$state, $timeout, $ionicScrollDelegate, chatPageFactory, $ionicLoading, $ionicHistory, timeStorage) {
+     function chatPageController($scope, $state, $timeout, $ionicScrollDelegate, chatPageFactory, $ionicLoading, $ionicHistory, timeStorage, socketService, $stateParams) {
          var self = this;
-         $ionicLoading.show();
-         $scope.userHeader ={};
+         $scope.userHeader = {};
          $scope.userHeader.name = 'kush';
          $scope.userHeader.image = 'http://lorempixel.com/640/480/nature';
          $scope.userHeader.lastSeen = 'LAST SEEN TODAY 08:03 PM';
-         $scope.userHeader.goBack = function(){
+         $scope.userHeader.goBack = function() {
              $ionicHistory.goBack();
          }
          $scope.userfooter = {};
          $scope.userfooter.image = 'http://lorempixel.com/640/480/nature';
+         socket.on('new_room_message', function(data) {
+             self.displayChatMessages.push({
+                 "image": data.image,
+                 "message": data.message,
+                 "messageTime": "2:12",
```

Comment:
Sir chatting with socket is not giving response time. So i used dummy for this. will change after when chatting fully configured.